analysis/src/xml.rs

```diff
@@ -0,0 +1,715 @@
+                                }
+                                None => {
+                                    registry.externals.push(External::from_node(type_node));
+                                }
+                                _ => (),
+                            }
+                        }
+                    }
+                }
+                "enums" => match registry_child.attribute("type") {
+                    Some("enum") => registry.enums.push(Enum::from_node(registry_child, api)),
+                    Some("bitmask") => registry
+                        .bitmasks
+                        .push(BitMask::from_node(registry_child, api)),
+                    _ => {
+                        if matches!(registry_child.attribute("name"), Some("API Constants")) {
+                            for enum_node in registry_child
+                                .children()
+                                .filter(|node| node.has_tag_name("enum"))
+                                .filter(|node| api_matches(node, api))
+                            {
+                                if enum_node.has_attribute("alias") {
+                                    registry.constant_aliases.push(Alias::from_node(enum_node));
+                                } else {
+                                    registry.constants.push(Constant::from_node(enum_node));
+                                }
+                            }
+                        }
+                    }
+                },
+                "commands" => {
+                    for command_node in registry_child
+                        .children()
+                        .filter(|node| node.has_tag_name("command"))
+                        .filter(|node| api_matches(node, api))
+                    {
+                        if command_node.has_attribute("alias") {
+                            registry
+                                .command_aliases
+                                .push(Alias::from_node(command_node));
+                        } else {
+                            registry
+                                .commands
+                                .push(Command::from_node(command_node, api));
+                        }
+                    }
+                }
+                "feature" => {
+                    registry
+                        .features
+                        .push(Feature::from_node(registry_child, api));
+                }
+                "extensions" => {
+                    for extension_node in registry_child
+                        .children()
+                        .filter(|node| node.has_tag_name("extension"))
+                        .filter(|node| {
+                            node.attribute("supported")
+                                .map(|values| values.split(',').any(|support| support == api))
+                                .unwrap_or(true)
```

Comment:
Deduplicate with `api_matches`? Only the attribute name is different.

Reply:
This is the only place in the code where we have to filter for the string "supported" instead of the much more common "api", so I decided not to disturb the `api_matches` function and just have a special case here. We could also have `api_matches` call another function `attribute_matches` with this code, and then use that here.

Reply:
It's not exactly the same as `api`, because it's value can also be `disabled`.

Side note, this also means that we automatically skip all disabled extensions as a consequence of this.

Reply:
> We could also have `api_matches` call another function `attribute_matches` with this code, and then use that here.

So, to be clear, did you suggest that this should be done?